sdk/identity/Azure.Identity/src/MsalConfidentialClient.cs

```diff
@@ -146,7 +146,7 @@ protected override async ValueTask<IConfidentialClientApplication> CreateClientA
 
             if (!string.IsNullOrEmpty(tenantId))
             {
-                builder.WithAuthority(Pipeline.AuthorityHost.AbsoluteUri, tenantId);
+                builder.WithTenantId(tenantId);
```

Comment:
This code path is hit by the ChallengeBasedAuthenticationPolicy in KeyVault. The `tenantId` comes from the challenge response. If `Pipeline.AuthorityHost` doesn't agree with the authority passed to the ConfidentialClientBuilder, then we hit the MSAL exception about mismatched authorities (https://aka.ms/msal-net-authority-override).

Switching to `.WithTenantId` sidesteps the issue of knowing what authority was passed to the ConfidentialClientBuilder.